opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java

```diff
@@ -186,9 +201,165 @@ private JSONObject createFeatureCollection() {
         return featureCollection;
     }
 
-    private ListTaskContract.Presenter getPresenter() {
-        return (ListTaskContract.Presenter) presenterCallBack;
+    public void saveJsonForm(String json) {
+        try {
+            JSONObject jsonForm = new JSONObject(json);
+            String encounterType = jsonForm.getString(ENCOUNTER_TYPE);
+            if (SPRAY_EVENT.equals(encounterType) || MOSQUITO_COLLECTION_EVENT.equals(encounterType) || LARVAL_DIPPING_EVENT.equals(encounterType)) {
+                saveMosquitoInterventionForm(jsonForm);
+            } else if (REGISTER_STRUCTURE_EVENT.equals(encounterType)) {
+                saveRegisterStructureForm(jsonForm);
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Error saving Json Form data", e);
+        }
     }
 
 
+    private org.smartregister.domain.db.Event saveEvent(JSONObject jsonForm, String encounterType, String bindType) throws JSONException {
+        String entityId = getString(jsonForm, ENTITY_ID);
+        JSONArray fields = JsonFormUtils.fields(jsonForm);
+        JSONObject metadata = getJSONObject(jsonForm, METADATA);
+
+        FormTag formTag = new FormTag();
+        formTag.providerId = sharedPreferences.fetchRegisteredANM();
+        formTag.locationId = sharedPreferences.fetchDefaultLocalityId(formTag.providerId);
+        formTag.teamId = sharedPreferences.fetchDefaultTeamId(formTag.providerId);
+        formTag.team = sharedPreferences.fetchDefaultTeam(formTag.providerId);
+        Event event = JsonFormUtils.createEvent(fields, metadata, formTag, entityId, encounterType, bindType);
+        JSONObject eventJson = new JSONObject(gson.toJson(event));
+        eventJson.put(DETAILS, getJSONObject(jsonForm, DETAILS));
+        eventClientRepository.addEvent(entityId, eventJson);
+        return gson.fromJson(eventJson.toString(), org.smartregister.domain.db.Event.class);
+    }
+
+    private void saveRegisterStructureForm(JSONObject jsonForm) {
```

Comment:
Remove this method, method already defined in the parent class  `BaseInteractor`

Reply:
Done.